docs/src/intro/usage.md

```diff
@@ -51,4 +52,138 @@ In the `miden/examples/fib` directory, we provide a very simple Fibonacci calcul
 ```
 ./target/release/miden run -a miden/examples/fib/fib.masm -n 1
 ```
-This will run the example code to completion and will output the top element remaining on the stack.
\ No newline at end of file
+This will run the example code to completion and will output the top element remaining on the stack.
+
+## REPL
+
+The Miden Read–eval–print loop or repl for short is a Miden shell that allows for quick and easy debugging with Miden assembly. When the repl gets initialised, there are a bunch of awesome things an user can do like execute any Miden instruction, undo executed instructions, check the state of the stack and memory at anytime and more! A `history.txt` file is saved when the repl is exited. One thing to note that all the repl native commands starts with an `!` to differentiate them with regular assembly instruction. The Repl currently supports the following commands:
+
+### Miden assembly instruction
+
+All the Miden instruction mentioned in the Miden Assembly document is valid. (Ex. push.0, drop, dropw, swap, cswap, u32checked_add, mem_loadw, mtree_get etc.). One can either input instructions one by one or multiple instructions in one input.
```

Comment:
A couple typos here:

* "All Miden instructions ..."
* "... document are valid." Also, the "document" should probably be "section" and we should probably use a link for it.